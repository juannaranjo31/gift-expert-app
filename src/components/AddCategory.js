import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if (inputValue.trim().length > 2) {
            setCategories( c => [inputValue, ...c]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input 
            type="text"
            value={inputValue}
            onChange={ (e) => setInputValue( e.target.value ) }
            placeholder='Ingrese una categoria para su busqueda de gifts :)'
            
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
