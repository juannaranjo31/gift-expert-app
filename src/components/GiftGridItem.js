import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ( {title, url} ) => {

    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}