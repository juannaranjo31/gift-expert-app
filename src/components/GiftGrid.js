import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifts } from '../hooks/useFetchGifts';
import { GiftGridItem } from './GiftGridItem';

export const GiftGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifts( category );

    return (
        <>
            <h3> {category} </h3>
            {loading && <p className='animate__animated animate__flash'>Loading...</p>}

            <div className='card-grid'>
                {
                    images.map((img) => {
                        return <GiftGridItem key={img.id} {...img} />
                    })
                }
            </div>

        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
};