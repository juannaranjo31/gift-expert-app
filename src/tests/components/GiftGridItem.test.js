import React from 'react'
import { shallow } from "enzyme";
import {GiftGridItem} from '../../components/GiftGridItem';

describe('Tests sobre el componenete <GiftGridItem />', () => {

    const title = 'un titulo';
    const url = 'https://localhost/algo.jpg';
    let wrapper = shallow(<GiftGridItem title={title} url={url} />);

    beforeEach( () => {
        wrapper = shallow(<GiftGridItem title={title} url={url} />);
    });

    //Snapshot correcto del componente
    test('El componente debe reenderizarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el titulo del Gif', () => {
        const p = wrapper.find('p');

        expect ( p.text().trim() ).toBe( title );
    });

    test('Debe tener props src igual a url y alt igual a titulo', () => {
        const img = wrapper.find('img');
        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title );
    });

    test('Debe tener una animacion fadeIn', () => {
        const div = wrapper.find('div');
        expect( div.prop('className').includes( 'animate__fadeIn' ) ).toBe( true );
    })
    
});
