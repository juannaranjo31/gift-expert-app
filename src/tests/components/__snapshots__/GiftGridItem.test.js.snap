// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests sobre el componenete <GiftGridItem /> El componente debe reenderizarse correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
